src/charm.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charmed operator for the SD-Core AMF service."""
+
+import logging
+from ipaddress import IPv4Address
+from subprocess import check_output
+
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires  # type: ignore[import]
+from charms.observability_libs.v1.kubernetes_service_patch import (  # type: ignore[import]
+    KubernetesServicePatch,
+)
+from charms.prometheus_k8s.v0.prometheus_scrape import (  # type: ignore[import]
+    MetricsEndpointProvider,
+)
+from charms.sdcore_nrf.v0.fiveg_nrf import NRFRequires  # type: ignore[import]
+from jinja2 import Environment, FileSystemLoader
+from lightkube.models.core_v1 import ServicePort
+from ops.charm import CharmBase, EventBase
+from ops.main import main
+from ops.model import ActiveStatus, BlockedStatus, MaintenanceStatus, WaitingStatus
+from ops.pebble import Layer
+
+logger = logging.getLogger(__name__)
+
+PROMETHEUS_PORT = 9089
+SBI_PORT = 29518
+NGAPP_PORT = 38412
+SCTP_GRPC_PORT = 9000
+AMF_DATABASE_NAME = "sdcore_amf"
+DEFAULT_DATABASE_NAME = "free5gc"
+CONFIG_DIR_PATH = "/free5gc/config"
```

Comment:
This path doesn't match the path in metadata.yaml, hence why the directory doesn't get greated.

Reply:
Good catch, thanks!